src/multipatch/connectivity/edge_transformation.hpp

```diff
@@ -12,6 +12,147 @@
+    /// @brief Grid parallel to Edge2 of the Interface.
+    using EdgeGrid2 = typename Interface::Edge2::parallel_grid;
+
+    /// @brief IdxRange parallel to the Edge on the current patch.
+    using IdxRangeCurrent = IdxRange<EdgeGrid2>;
+    /// @brief IdxRange parallel to the Edge on the target patch.
+    using IdxRangeTarget = IdxRange<EdgeGrid1>;
+};
+
+/**
+ * @brief Specification for current continuous dimension being the 
+ * continuous dimension of the grid on Edge1 of the given Interface. 
+ */
+template <class Interface>
+struct IdxRangeCurrentAndTarget<
+        Interface,
+        typename Interface::Edge1::parallel_grid::continuous_dimension_type>
+{
+    /// @brief Grid parallel to Edge1 of the Interface.
+    using EdgeGrid1 = typename Interface::Edge1::parallel_grid;
+    /// @brief Grid parallel to Edge2 of the Interface.
+    using EdgeGrid2 = typename Interface::Edge2::parallel_grid;
+
+    /// @brief IdxRange parallel to the Edge on the current patch.
+    using IdxRangeCurrent = IdxRange<EdgeGrid1>;
+    /// @brief IdxRange parallel to the Edge on the target patch.
+    using IdxRangeTarget = IdxRange<EdgeGrid2>;
+};
+
+/**
+ * @brief Specification for current continuous dimension being the 
+ * continuous dimension of the grid on Edge2 of the given Interface. 
+ */
+template <class Interface>
+struct IdxRangeCurrentAndTarget<
+        Interface,
+        typename Interface::Edge2::parallel_grid::continuous_dimension_type>
+{
+    /// @brief Grid parallel to Edge1 of the Interface.
+    using EdgeGrid1 = typename Interface::Edge1::parallel_grid;
+    /// @brief Grid parallel to Edge2 of the Interface.
+    using EdgeGrid2 = typename Interface::Edge2::parallel_grid;
+
+    /// @brief IdxRange parallel to the Edge on the current patch.
+    using IdxRangeCurrent = IdxRange<EdgeGrid2>;
+    /// @brief IdxRange parallel to the Edge on the target patch.
+    using IdxRangeTarget = IdxRange<EdgeGrid1>;
+};
+
+
+/**
+ * @brief Get equivalent coordinate on the target patch. 
+ * 
+ * @tparam PatchOrDim Type for a continuous dimension or a patch. 
+ * @tparam Interface Interface type given in EdgeTransformation class. 
+ * @tparam CurrentDim The continuous dimension where the given coordinate is defined. 
+ * 
+ * @param current_coord Coordinate on the current patch. 
+ * @param idx_range_patch_1 Index range on the Edge1 of the Interface. 
+ * @param idx_range_patch_2 Index range on the Edge2 of the Interface. 
+ * 
+ * @return Equivalent coordinate on the target patch. 
+ */
+template <typename PatchOrDim, class Interface, class CurrentDim>
+Coord<std::conditional_t<
+        std::is_same_v<
+                CurrentDim,
+                typename Interface::Edge1::parallel_grid::continuous_dimension_type>,
+        typename Interface::Edge2::parallel_grid::continuous_dimension_type,
+        typename Interface::Edge1::parallel_grid::continuous_dimension_type>>
+get_equivalent_target_coordinate(
```

Comment:
I think the use of `PatchOrDim` makes this harder to follow. Can you not ask for the Patch here and find it in `operator()` if you don't have it yet